backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/ActivitiesManager.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.impl;
+
+import com.github.jknack.handlebars.internal.lang3.StringUtils;
+import io.micronaut.core.annotation.Nullable;
+import org.hyperledger.aries.api.connection.ConnectionState;
+import org.hyperledger.aries.api.issue_credential_v1.CredentialExchangeState;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeRole;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeState;
+import org.hyperledger.bpa.controller.api.activity.*;
+import org.hyperledger.bpa.model.Partner;
+import org.hyperledger.bpa.model.PartnerProof;
+import org.hyperledger.bpa.repository.BPACredentialExchangeRepository;
+import org.hyperledger.bpa.repository.PartnerProofRepository;
+import org.hyperledger.bpa.repository.PartnerRepository;
+
+import javax.inject.Inject;
+import java.util.*;
+
+public class ActivitiesManager {
```

Comment:
This class is hard to understand and to read. I guess one part is static config (that should not be static, see comment below) that decides when an event is triggered in the handler. So there is a trigger section. The other part is about mapping the persisted objects back to these events to display in the UI. 

One or all of the following might help to make this class more readable:

1. Add this description as a comment
2. Split this up into a trigger and read section and rename the methods accordingly

What do you think?

Reply:
Configuration and lists are moved out into `ActivityLogConfig` class, so the `ActivitiesManager` is now just reading the db and returning the results.